pallets/dkg-proposal-handler/src/tests.rs

```
@@ -1,7 +1,191 @@
+	execute_test_with(|| {
+		// First submission
+		let tx_v_2 = TransactionV2::EIP2930(mock_eth_tx_eip2930(0));
+
+		assert_ok!(DKGProposalHandler::handle_proposal(tx_v_2.encode(), ProposalAction::Sign(0)));
+		assert_eq!(DKGProposalHandler::unsigned_proposals(0, 0).is_some(), true);
+
+		let signed_proposal = mock_signed_proposal(tx_v_2.clone());
+
+		assert_ok!(DKGProposalHandler::submit_signed_proposal(
+			Origin::root(),
+			signed_proposal.clone()
+		));
+
+		assert_eq!(DKGProposalHandler::unsigned_proposals(0, 0).is_none(), true);
+		assert_eq!(DKGProposalHandler::signed_proposals(0, 0).is_some(), true);
+
+		// Second submission
+		assert_ok!(DKGProposalHandler::handle_proposal(tx_v_2.encode(), ProposalAction::Sign(0)));
+		assert_eq!(DKGProposalHandler::unsigned_proposals(0, 0).is_some(), true);
+
+		assert_ok!(DKGProposalHandler::submit_signed_proposal(Origin::root(), signed_proposal));
+
+		assert_eq!(DKGProposalHandler::unsigned_proposals(0, 0).is_none(), true);
+		assert_eq!(DKGProposalHandler::signed_proposals(0, 0).is_some(), true);
+	});
+}
+
+#[test]
+fn submit_signed_proposal_fail_invalid_sig() {
+	execute_test_with(|| {
+		let tx_eip2930 = EIP2930Transaction {
+			chain_id: 0,
+			nonce: U256::from(0u8),
+			gas_price: U256::from(0u8),
+			gas_limit: U256::from(0u8),
+			action: TransactionAction::Create,
+			value: U256::from(99u8),
+			input: Vec::<u8>::new(),
+			access_list: Vec::new(),
+			odd_y_parity: false,
+			r: H256::from([0u8; 32]),
+			s: H256::from([0u8; 32]),
+		};
+		let tx_v_2 = TransactionV2::EIP2930(tx_eip2930);
+
+		assert_ok!(DKGProposalHandler::handle_proposal(tx_v_2.encode(), ProposalAction::Sign(0)));
+		assert_eq!(DKGProposalHandler::unsigned_proposals(0, 0).is_some(), true);
+
+		let mut invalid_sig: Vec<u8> = Vec::new();
+		invalid_sig.extend_from_slice(&[0u8, 64]);
+		let signed_proposal =
+			ProposalType::EVMSigned { data: tx_v_2.encode(), signature: invalid_sig };
+
+		assert_eq!(
```

This seems like it should also fail due to invalid sig.